### DISCUSSION TITLE
安装了nvm-desktop自动把我的环境变量改成了地址

### DISCUSSION THREAD

**Discussion by jingtianyuke** [1111mp/nvm-desktop#45] [Ideas]
![image](https://github.com/1111mp/nvm-desktop/assets/122524331/e26222b6-8168-4f71-a8dc-14d89fda70f2)
安装完之后会篡改我的path的值

**Comment by 1111mp** [comment]
首先 `nvm-desktop` 并不会篡改系统任何东西的 只不过为了让用户方便 会在第一次安装的时候（通过`"%HOMEPATH%\.nvmd\migration"`文件来做版本控制）通过命令往系统环境变量写入`"%HOMEPATH%\.nvmd\bin"` 这样用户就不需要手动添加了（目前没有实现卸载的时候通过脚本移除该变量）具体的代码可以查看：[src/main/utils/migration.ts#L148](https://github.com/1111mp/nvm-desktop/blob/228e954035f1622604e2ba7d53588ce793a68a5e/src/main/utils/migration.ts#L148)

不过有一种可能 就是有些环境变量是在某个`user`下的，但是你当前是`admin`用户，然后通过这个命令`setx -m PATH`添加环境变量确实会造成将`user`下的变量也添加到`system`下面了。

**Reply by jingtianyuke** [reply]
感谢你的解答，最近我使用这个node.js版本控制工具无法改变我的npm的版本了，可能是我上次指定了某个npm的版本导致无法的自由切换了有什么的办法解决吗

**Reply by 1111mp** [reply]
每个node下会有一个默认的npm版本（可以在列表里面看到），然后你可以通过 npm install npm -g 来将npm更新到最新版本，npm install npm@version -g 来安装到具体的版本。 或者你也可以使用corepack来管理，具体的命令你可以去官方文档查看一下，可以用来切换版本（npm tarn pnpm都可）

**Reply by jingtianyuke** [reply]
非常感谢您的回答
